problem_builder/mixins.py

```diff
@@ -185,19 +184,28 @@ def studio_view(self, _context=None):
 
 class StudentViewUserStateMixin(object):
     """
-    Mixin to provide student_view_user_state view
+    Mixin to provide student_view_user_state view.
+
+    To prevent unnecessary overloading of the build_user_state_data method,
+    you may specify `USER_STATE_FIELDS` to customise build_user_state_data
+    and student_view_user_state output.
     """
     NESTED_BLOCKS_KEY = "components"
     INCLUDE_SCOPES = (Scope.user_state, Scope.user_info, Scope.preferences)
 
+    # Default field names to include in build_user_state_data
+    USER_STATE_FIELDS = ['completed', 'num_attempts', 'student_results']
```

Comment:
I think it could potentially be confusing to have a set of default fields like this.